src/joystick/hidapi/SDL_hidapi_8bitdo.c

```diff
@@ -0,0 +1,431 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+#ifdef SDL_JOYSTICK_HIDAPI
+
+#include "../SDL_sysjoystick.h"
+#include "SDL_hidapijoystick_c.h"
+#include "SDL_hidapi_rumble.h"
+
+#ifdef SDL_JOYSTICK_HIDAPI_8BITDO
+
+// Define this if you want to log all packets from the controller
+#if 0
+#define DEBUG_8BITDO_PROTOCOL
+#endif
+
+enum
+{
+    SDL_GAMEPAD_BUTTON_8BITDO_L4 = 12,
```

Comment:
It looks like there's no way to generate button 11, so maybe L4 should start there instead?
```suggestion
    SDL_GAMEPAD_BUTTON_8BITDO_L4 = 11,
```